File: backend/src/Domain/Models/Severity.cs
namespace backend.src.Domain.Models;

public class Severity
{
    public int SeverityId { get; set; }          // PK
    public int? QuestionId { get; set; }         // FK -> Question (nullable)
    public int? MeasurementId { get; set; }      // FK -> Measurement (nullable)
    public int? RequiredOption { get; set; }     // FK -> QuestionOption
    public string? RequiredText { get; set; }
    public decimal? RequiredValue { get; set; }
    public string Operator { get; set; } = "";
    public int Score { get; set; }

    public Question? Question { get; set; }
    public Measurement? Measurement { get; set; }
    public QuestionOption? RequiredOptionNavigation { get; set; }
}

File: frontend/src/types/Severity.ts
export type CreateSeverityDto = {
  questionId?: number | null;
  measurementId?: number | null;
  requiredOption?: number | null;
  requiredText?: string | null;
  requiredValue?: number | null;
  operator: string;
  score: number;
};

export type SeverityDto = {
  severityId: number;
  questionId?: number | null;
  measurementId?: number | null;
  requiredOption?: number | null;
  requiredText?: string | null;
  requiredValue?: number | null;
  operator: string;
  score: number;
};
